src/providers/oauth-provider.js

```diff
@@ -26,27 +26,21 @@ var requiredKeys = [
  */
 
 function OAuthProvider() {
-  var config;
+  var defaultConfig;
```

Comment:
@lionelB it looks quite good overall. However I can see some issues here. It might introduce a memory leak, because you're creating a closure. In order to avoid this kind of issues I would some trivial changes, and I would recommend to assign `defaultConfig` to `this` such as:
```js
this.defaultConfig = null;
```
I'll continue commenting below to show how to fix this.